### PR TITLE
Ability to add listed attorney claimant

### LINKED ISSUE #15567: Non-Veteran Claimants | Ability to add Attorney claimant listed in CorpDB
Note: This is being replaced by [Jira-628](https://vajira.max.gov/browse/CASEFLOW-628)

Keeping the old AC in case it's useful at some point, but the ticket breakout and AC are different in the new Jira tickets.

## User or job story
As a BVA Caseflow Intake user, I should be able to add an attorney found in CorpDB as a claimant to a board appeal.

This ticket handles the ability to add a listed Attorney claimant if that relationship option is selected.  This will be building off of previous work to add attorney claimants when the claimant is found.

## Acceptance criteria
- [ ] Please put this work behind the feature toggle: [non_veteran_claimants]
- [ ] On the "Add Claimant" page, if the user selects the "Attorney" relationship option, show the ability to search for a claimant listed in the Corp DB to add
- [ ] If an attorney is found and selected, show their address below the dropdown (see design)
- [ ] If the user selects a listed claimant, enable the button to Continue, and move to the Add Issues screen with the selected claimant added
- [ ] Include screenshot(s) in the Github issue if there are front-end changes

### Out of scope
- Unlisted Attorney claimant

### Designs
Note: This is design is from adding a POA, but design will be similar for adding an attorney

### Technical notes
The ability to handle listed attorney claimants is currently available through another workflow.

### PR DESCRIPTION
**Pull request by leikkisa**
Resolves #15567

[Jira-628](https://vajira.max.gov/browse/CASEFLOW-628)

### Description
Adds the ability to add a listed attorney claimant to the unrecognized appellant flow.

### Acceptance Criteria
- [x] Given that an Intake User is on the Add Claimant Page, when the user selects "Attorney" Relationship type, then the user is prompted to select attorneys from the drop down list. 
- [x] Given that an Attorney does exist in the CorpDB, when the user selects a name listed from the drop down, then the user is able to see the Attorneys information and answer the 21-22 prompt. 
- [x] Given that an Intake User has selected an existing Attorney, when the user answers the 21-22 prompt, then the user is able to click the button "Continue to Next Step".
- [x] Given that an Attorney does not exist in the CorpDB, when the user selects Name Not Listed, then the user is able to select if the claimant is an Individual or Organization.
- [x] Given that the claimant is an Individual Attorney, when the user fills out the required fields and has selected a response to the 21-22 prompt, then the user is able to click the button "Continue to Next Step".
- [x] Given that the claimant is an Organization Attorney, when the user fills out the required fields and has selected a response to the 21-22 prompt, then the user is able to click the button "Continue to Next Step".
- [x] Given that the Intake User has entered in claimant data, when the user does not fill out the required fields or has not selected an answer to the 21-22 prompt, then the user is not able to click the button "Continue to Next Step"
- [x] Given that an Intake User has completed all the required fields on the Add Claimant Page, when the intake user clicks "Continue To Next Step", then the user is not redirected to new page. 
- [x] Given that an Intake User is on the Add Claimant Page, if the user clicks the back button, then the user is sent back to the Review page with all information populated that was submitted prior. 

### Testing Plan
1. Feature test and local testing

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

### CODE REVIEW

File: client/app/intake/addClaimant/AddClaimantPage.jsx [hunk 1]
```diff
@@ -75,6 +115,40 @@ export const AddClaimantPage = () => {
             as={SearchableDropdown}
           />
           <br />
+          { watchRelationship === 'attorney' &&
+            <>
+              <Controller
+                control={control}
+                name="listedAttorney"
+                defaultValue=""
```
Comment (leikkisa) [inline client/app/intake/addClaimant/AddClaimantPage.jsx:123]: Wasn't sure about the default value part, but was seeing some messages asking for it.

**Comment by jcq** [comment]
Just to clarify, is handling submit fully out of scope for this? The ticket AC seemed to indicate that both enabling the continue button and moving to the next step were part of this, but that might have changed

**Comment by leikkisa** [comment]
@jcq I copy and pasted this AC from Jira:
> Given that an Intake User has completed all the required fields on the Add Claimant Page, when the intake user clicks "Continue To Next Step", then the user is not redirected to new page.

**Comment by leikkisa** [comment]
Screen Shot 2021-01-27 at 2 23 57 PM
Screen Shot 2021-01-27 at 2 23 48 PM
Screen Shot 2021-01-27 at 2 23 40 PM
Screen Shot 2021-01-27 at 2 23 28 PM

**Comment by jcq** [comment]
Also, adjusting AddClaimantPage.stories.js thusly should at least set the progress bar to what is expected:

```jsx
import { PAGE_PATHS } from '../constants';

const RouterDecorator = (Story) => (
  <MemoryRouter initialEntries={[PAGE_PATHS.ADD_CLAIMANT]}>
    <Story />
  </MemoryRouter>
);
```

**Review by jcq — APPROVED** [review]
Looks good once tests pass!

File: client/app/intake/addClaimant/utils.js [hunk 2]
```diff
@@ -34,16 +37,28 @@ export const schema = yup.object().shape({
...
+	  is: (value) => ['individual', 'organization'].includes(value),
+    then: yup.number().min(5).required()
+  }),
+  country: yup.string().when('partyType', {
+	  is: (value) => ['individual', 'organization'].includes(value),
+    then: yup.string().required()
+  }),
   email: yup.string().email(),
   phoneNumber: yup.string(),
   vaForm: yup.string().required(),
```
[the remark this answers is not in the document]
Comment (leikkisa) [reply client/app/intake/addClaimant/utils.js:64]: Good catch!

[the remark this answers is not in the document]
Comment (leikkisa) [reply client/app/intake/addClaimant/utils.js:64]: Nevermind, this is always required, even though it's conditionally displayed, ultimately it gets displayed in every scenario (once the preceding selections are made)

**Pull request merged**